[BUG] schema decorator does not play nice with pytest marks

**Checklist**

- [x] I checked the [FAQ section](https://schemathesis.readthedocs.io/en/stable/faq.html#frequently-asked-questions) of the documentation
- [x] I looked for similar issues in the [issue tracker](https://github.com/schemathesis/schemathesis/issues)

**Describe the bug**
Because schemathesis can be quite slow and requires a testing version of the API to be set up, we have it behind `pytest.mark.acceptance` to avoid running it with unit tests. Unfortunately, the schema initializes when we set the schema variable, rather than when the test runs. To get around that, we use `schemathesis.from_pytest_fixture` to get the schema. When the `mark.acceptance` is above `schema.parametrize`, the tests hang forever. When it is below `schema.parametrize`, the schema is evaluated and errors because we don't have a test API running.

**To Reproduce**
Steps to reproduce the behavior:
```python
@pytest.fixture
def web_app():
    return schemathesis.from_uri(...)

schema = schemathesis.from_pytest_fixture("web_app")

@pytest.mark.acceptance
@schema.parametrize()
def test_schema(case):
    ...
```
When this is run, tests will hang forever. When the decorators on the test are reversed and the test API is not running, tests will fail because the test API is not running.

This does not depend on the API schema, since the whole point is that the schema is not running.

**Expected behavior**
When the acceptance mark is deselected in the example above, the tests should pass. When the acceptance mark is selected in the example above, the tests should not stall out (and also, eventually, pass, as long as the test API is running in that context.)

**Environment (please complete the following information):**
 - OS: Debian docker container
 - Python version: 3.8.5
 - Schemathesis version: 3.12.3
 - Spec version: NA

**Additional context**
Any other workaround that allows us to skip this test with marks in a sane way would also be a welcome solution.

Hi @schicks 

Sorry for not getting back to you earlier. What is the `pytest` version you're using? I created a similar test file and can't reproduce the behavior - the `web_app` fixture is not evaluated, and the relevant test is skipped as expected when I use `-m "not acceptance"`:

```python
import pytest

import schemathesis


@pytest.fixture
def web_app():
    1 / 0


schema = schemathesis.from_pytest_fixture("web_app")


@pytest.mark.acceptance
@schema.parametrize()
def test_schema(case):
    1 / 0
```

Deselecting:
```
pytest -m "not acceptance" example.py
=================================== test session starts ===================================
platform linux -- Python 3.10.4, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /tmp
plugins: mock-1.13.0, schemathesis-3.13.3, httpserver-1.0.3, xdist-1.34.0, forked-1.4.0, hypothesis-6.41.0, anyio-3.4.0, subtests-0.5.0, asyncio-0.11.0
collected 1 item / 1 deselected                                                           

==================================== warnings summary =====================================
example.py
  /tmp/example.py:14: PytestUnknownMarkWarning: Unknown pytest.mark.acceptance - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.acceptance

-- Docs: https://docs.pytest.org/en/stable/warnings.html
============================ 1 deselected, 1 warning in 0.01s =============================
```

Selecting:

```
 pytest -m "acceptance" /tmp/example.py 
=================================== test session starts ===================================
platform linux -- Python 3.10.4, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /tmp
plugins: mock-1.13.0, schemathesis-3.13.3, httpserver-1.0.3, xdist-1.34.0, forked-1.4.0, hypothesis-6.41.0, anyio-3.4.0, subtests-0.5.0, asyncio-0.11.0
collected 1 item                                                                          

example.py E                 [100%]

========================================= ERRORS ==========================================
______________________________ ERROR at setup of test_schema ______________________________

    @pytest.fixture
    def web_app():
>       1 / 0
E       ZeroDivisionError: division by zero

/tmp/example.py:8: ZeroDivisionError
==================================== warnings summary =====================================
example.py:14
  /tmp/example.py:14: PytestUnknownMarkWarning: Unknown pytest.mark.acceptance - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.acceptance

-- Docs: https://docs.pytest.org/en/stable/warnings.html
================================= short test summary info =================================
ERROR ...
=============================== 1 warning, 1 error in 0.08s ===============================
```

Environment:
- Python 3.10.4
- pytest 6.2.5
- latest available Schemathesis version

Though, when the decorators' order is reversed, they are indeed ignored :( It should be solved by #1441 

Am I missing something?

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -21,9 +21,10 @@ Changelog
 
 **Fixed**
 
-- Missing ``source`` attribute in the ``Case.partial_deepcopy`` implementation. `#1429`
+- Missing ``source`` attribute in the ``Case.partial_deepcopy`` implementation. `#1429`_
 - Duplicated failure message from ``content_type_conformance`` and ``response_schema_conformance`` checks when the checked response has no ``Content-Type`` header. `#1394`_
 - Not copied ``case`` & ``response`` inside ``Case.validate_response``.
+- Ignored ``pytest.mark`` decorators when they are applied before ``schema.parametrize`` if the schema is created via ``from_pytest_fixture``. `#1378`_
 
 `3.13.8`_ - 2022-04-05
 ----------------------
@@ -2400,6 +2401,7 @@ Deprecated
 .. _#1410: https://github.com/schemathesis/schemathesis/issues/1410
 .. _#1395: https://github.com/schemathesis/schemathesis/issues/1395
 .. _#1394: https://github.com/schemathesis/schemathesis/issues/1394
+.. _#1378: https://github.com/schemathesis/schemathesis/issues/1378
 .. _#1372: https://github.com/schemathesis/schemathesis/issues/1372
 .. _#1370: https://github.com/schemathesis/schemathesis/issues/1370
 .. _#1366: https://github.com/schemathesis/schemathesis/issues/1366

--- a/src/schemathesis/lazy.py
+++ b/src/schemathesis/lazy.py
@@ -1,5 +1,5 @@
 from inspect import signature
-from typing import Any, Callable, Dict, Optional, Union
+from typing import Any, Callable, Dict, Generator, Optional, Union
 
 import attr
 import pytest
@@ -110,11 +110,7 @@ class LazySchema:
                 if not tests:
                     fail_on_no_matches(node_id)
                 request.session.testscollected += len(tests)
-                capmam = request.node.config.pluginmanager.get_plugin("capturemanager")
-                if capmam is not None:
-                    suspend_capture_ctx = capmam.global_and_fixture_disabled
-                else:
-                    suspend_capture_ctx = nullcontext
+                suspend_capture_ctx = _get_capturemanager(request)
                 subtests = SubTests(request.node.ihook, suspend_capture_ctx, request)
                 for result, data_generation_method in tests:
                     if isinstance(result, Ok):
@@ -126,6 +122,7 @@ class LazySchema:
                 subtests.item._nodeid = node_id
 
             test = pytest.mark.usefixtures(self.fixture_name)(test)
+            _copy_marks(func, test)
 
             # Needed to prevent a failure when settings are applied to the test function
             test.is_hypothesis_test = True  # type: ignore
@@ -136,6 +133,19 @@ class LazySchema:
 
     def given(self, *args: GivenInput, **kwargs: GivenInput) -> Callable:
         return given_proxy(*args, **kwargs)
+
+
+def _copy_marks(source: Callable, target: Callable) -> None:
+    marks = getattr(source, "pytestmark", [])
+    # Pytest adds this attribute in `usefixtures`
+    target.pytestmark.extend(marks)  # type: ignore
+
+
+def _get_capturemanager(request: FixtureRequest) -> Generator:
+    capturemanager = request.node.config.pluginmanager.get_plugin("capturemanager")
+    if capturemanager is not None:
+        return capturemanager.global_and_fixture_disabled
+    return nullcontext
 
 
 def _get_node_name(node_id: str, operation: APIOperation, data_generation_method: DataGenerationMethod) -> str:

--- a/test/test_lazy.py
+++ b/test/test_lazy.py
@@ -1,8 +1,5 @@
 import pytest
 
-from schemathesis import DataGenerationMethod
-from schemathesis.constants import IS_PYTEST_ABOVE_54
-
 
 def test_default(testdir):
     # When LazySchema is used
@@ -689,3 +686,33 @@ def test_(case):
             r"match any API operations and therefore has no effect"
         ]
     )
+
+
+@pytest.mark.parametrize(
+    "decorators",
+    (
+        """@schema.parametrize()
+@pytest.mark.acceptance""",
+        """@pytest.mark.acceptance
+@schema.parametrize()""",
+    ),
+)
+def test_marks_transfer(testdir, decorators):
+    # See GH-1378
+    # When a pytest mark decorator is applied
+    testdir.make_test(
+        f"""
+@pytest.fixture
+def web_app():
+    1 / 0
+
+schema = schemathesis.from_pytest_fixture("web_app")
+
+{decorators}
+def test_schema(case):
+    1 / 0
+    """
+    )
+    result = testdir.runpytest("-m", "not acceptance")
+    # Then deselecting by a mark should work
+    result.assert_outcomes()

